pkg/k8shandler/collection.go

```diff
@@ -323,6 +329,25 @@ func updateCurrentDaemonsetImages(current *apps.DaemonSet, desired *apps.DaemonS
 	return current
 }
 
+func areResourcesDifferent(current *apps.Daemonset, desired *apps.Daemonset) (bool, *apps.Daemonset) {
```

Comment:
you could have a single `areResourcesDifferent` function - use introspection to determine if object is a Daemonset or a Deployment or a CronJob or whatever